src/ralph/admin/suggest.py

```diff
@@ -0,0 +1,77 @@
+import operator
+from functools import reduce
+
+from django.db.models import Q
+from django.http import JsonResponse
+from django.views.generic import View
+
+QUERY_PARAM = 'q'
+
+
+class JsonViewMixin(object):
+    """
+    A mixin that can be used to render a JSON response.
+    """
+    def render_to_json_response(self, context, **response_kwargs):
+        """
+        Returns a JSON response, transforming 'context' to make the payload.
+        """
+        return JsonResponse(self.get_data(context), **response_kwargs)
+
+    def get_data(self, context):
+        """
+        Returns an object that will be serialized as JSON by json.dumps().
+        """
+        return context
+
+
+class SuggestView(JsonViewMixin, View):
+    http_method_names = ['get']
+
+    def get(self, request, *args, **kwargs):
+        """
+        Returns serialized dict as JSON object.
+        """
+        return self.render_to_json_response({
+            'results': [
+                {'pk': obj.pk, '__str__': str(obj)}
+                for obj in self.get_queryset()
+            ]
+        })
+
+
+class AjaxSuggestMixin(object):
+    def get_urls(self):
+        from django.conf.urls import url
```

Comment:
it used here https://github.com/allegro/ralph/pull/1627/files#diff-73c1233d9acef15d6945e388915ad4f4R71

I will move this import block to top.